docs/source/AdministratorGuide/ServerInstallations/InstallingWebAppDIRAC.rst

```diff
@@ -256,154 +257,219 @@ Install and configure NGINX
+     ssl_session_tickets off;
 
-      # WebSocket support (nginx 1.4)
-      proxy_http_version 1.1;
-      proxy_set_header Upgrade $http_upgrade;
-      proxy_set_header Connection "upgrade";
+     # Diffie-Hellman parameter for DHE ciphersuites, recommended 2048 bits
+     # Generate your DH parameters with OpenSSL:
+     # ~ cd /etc/nginx/ssl
+     # ~ openssl dhparam -out dhparam.pem 4096 
+     ssl_dhparam /etc/nginx/ssl/dhparam.pem;
 
-      break;
-     }
-     location / {
-      rewrite ^ http://$server_name/DIRAC/ permanent;
-      }
-    }
-   server {
-     listen 443 default ssl; ## listen for ipv4
+     # HSTS (ngx_http_headers_module is required) (15768000 seconds = 6 months)
+     add_header Strict-Transport-Security max-age=15768000;
 
-     #server_name  lbvobox33.cern.ch;
-     server_name  dzmathe.cern.ch;
+     # To secure NGINX from Click-jacking attack
+     add_header X-Frame-Options SAMEORIGIN always;
 
-     ssl_prefer_server_ciphers On;
-     ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
-     ssl_ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS;
+     # OCSP Stapling --- fetch OCSP records from URL in ssl_certificate and cache them
+     ssl_stapling on;
+     ssl_stapling_verify on;
 
-     #Certs that will be shown to the user connecting to the web.
-     #Preferably NOT grid certs. Use something that the user cert will not complain about
-     ssl_certificate    /opt/dirac/etc/grid-security/hostcert.pem;
-     ssl_certificate_key /opt/dirac/etc/grid-security/hostkey.pem;
+     # DNS resolver for stapling so that the resolver defaults to Google’s DNS
+     resolver 8.8.4.4 8.8.8.8;
 
      ssl_client_certificate /opt/dirac/pro/etc/grid-security/cas.pem;
-   #  ssl_crl /opt/dirac/pro/etc/grid-security/allRevokedCerts.pem;
-     ssl_verify_client on;
+     # ssl_crl /opt/dirac/pro/etc/grid-security/allRevokedCerts.pem;
+     ssl_verify_client optional;
      ssl_verify_depth 10;
      ssl_session_cache shared:SSL:10m;
 
      root /opt/dirac/pro;
 
+    # The same directory must be exist with 'rw' permissions for all
+    location /files {
+      # Access for GET requests without certificate
+      if ($request_method = GET) {
+        # Webdav sever
+        error_page 418 = @webdav;
+        return 418;
+      }
+
+      # For not GET requests access only with client certificate verification
+      if ($ssl_client_verify = NONE) {
+        return 403 'certificate not found';
+      }
+      if ($ssl_client_verify != SUCCESS) {
+        return 403 'certificate verify failed';
+      }
+
+      # Webdav sever
+      error_page 418 = @webdav;
+      return 418;
+    }
+
+    location @webdav {
+        satisfy any;
+        # Read access for all
+        limit_except GET {
+          # Add allowed IP for not GET requests
+          #allow XXX.XXX.XXX.XXX;
+          deny  all;
+        }
```

Comment:
Do I understand correctly that the configuration above means "no-one can POST/PUT/DELETE"?
If this is the case, then we should suggest to add the IP host used by the master CS.

Reply:
yes, fixed